File: src/at/jku/ssw/coco/intellij/psi/CocoFile.java
package at.jku.ssw.coco.intellij.psi;

import at.jku.ssw.coco.intellij.CocoFileType;
import at.jku.ssw.coco.intellij.CocoLanguage;
import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

/**
 * @author Thomas Scheinecker <a href="mailto:tscheinecker@gmail.com">tscheinecker@gmail.com</a>
 */
public class CocoFile extends PsiFileBase {
    public CocoFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, CocoLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return CocoFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Coco File";
    }
}

File: src/main/java/at/scheinecker/intellij/coco/psi/CocoFile.kt
package at.scheinecker.intellij.coco.psi

import at.scheinecker.intellij.coco.CocoFileType
import at.scheinecker.intellij.coco.CocoLanguage
import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

/**
 * @author Thomas Scheinecker [tscheinecker@gmail.com](mailto:tscheinecker@gmail.com)
 */
class CocoFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, CocoLanguage.INSTANCE) {

    override fun getFileType(): FileType {
        return CocoFileType.INSTANCE
    }

    override fun toString(): String {
        return "Coco File"
    }
}
